design/host/host_db_requirements.md

```diff
@@ -0,0 +1,126 @@
+# Moving Host DB to SQL 
+
+The current implementation uses the `ethdb.KeyValueStore` which provides fast access but is not sufficient for the 
+querying capabilities required by Tenscan. We want to move to an SQL implementation similar to what the Enclave uses.
+
+## Current Storage 
+| Data Type        | Key             | Value                                     |
+|------------------|-----------------|-------------------------------------------|
+| **Batch**        | Batch Hash      | Batch Header                              |
+| **Batch**        | Batch Hash      | Transaction Hashes                        |
+| **Batch**        | Batch Hash      | Batch Data                                |
+| **Batch**        | Batch Hash      | Transaction hashes belonging to the Batch |
+| **Block**        | L1 Block Hash   | L1 Block Header                           |
+| **Block**        | L1 Block Number | L1 Block Header                           |
+| **Rollup**       | L1 Block Hash   | Rollup Header                             |
+| **Rollup**       | Rollup Hash     | Rollup Header                             |
+| **Transactions** | Transactions    | Total number of transactions              |
+
+## Tenscan Functionality Requirements
```

Comment:
I think it's worth creating 2 lists. 

First for the mainnet features.
And second for the extra features we want on testnet.

Reply:
Have added a todo section for the xchain messaging and mainnet vs testnet since we can quickly hash this out next week in person

Reply:
I was thinking in terms of the "decryption" functionality, which is testnet specific.
This is a testnet feature:
- Decrypt the encrypted TX blob